graphql/handler/apollofederatedtracingv1/tracing.go

```diff
@@ -8,18 +8,35 @@ import (
 	"google.golang.org/protobuf/proto"
 
 	"github.com/99designs/gqlgen/graphql"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+)
+
+const (
+	ERROR_MASKED     = "masked"
+	ERROR_UNMODIFIED = "all"
+	ERROR_TRANSFORM  = "transform"
 )
 
 type (
 	Tracer struct {
 		ClientName string
 		Version    string
 		Hostname   string
+		Errors     TraceErrors
 	}
 
 	treeBuilderKey string
 )
 
+type TraceErrors struct {
+	// ErrorOption is the option to handle errors in the trace, it can be one of the following:
+	// - "masked": masks all errors
+	// - "all": includes all errors
+	// - "transform": includes all errors but transforms them using TransformFunction, which can allow users to redact sensitive information
+	ErrorOption       string
+	TransformFunction func(gqlerror.Error) gqlerror.Error
```

Comment:
The `gqlerror.Error` struct you are passing around by values can be heavy, so I would prefer you use pointers where possible.
```suggestion
	TransformFunction func(*gqlerror.Error) *gqlerror.Error
```